cmp/grid/Grid.js

```diff
@@ -489,6 +498,14 @@ export class Grid extends Component {
         };
     }
 
+    // Get the number of records to be removed in upcoming update. For AG-2879 Workaround.
+    pendingDeletionCount(newRecords, api) {
+        const ids = new Set();
+        api.forEachNode((node, index) => ids.add(node.id));
+        newRecords.forEach(rec => ids.delete(rec.id));
+        return ids.size;
```

Comment:
Obviously depending on these loops and especially `Set` being fast, which I am sure they are. 

So ironic that we're doing here exactly what ag-grid does that is so slow - building up a flat list then winnowing it down one-by-one - but the use of Set makes all the difference.  I wonder how babel transpiles this code - I confess I would need to load it back in my brain, but is it relying on the browser supporting Set or replacing with a different backing implementation?